internal/helper/helper.go

```diff
@@ -284,6 +284,68 @@ func SafeQuantitySubtraction(orig, subtract quantity.Size) quantity.Size {
 	return orig - subtract
 }
 
+// CreateTarArchive places all of the files from a source directory into a tar.
+// Currently supported are uncompressed tar archives and the following
+// compression types: zip, gzip, xz bzip2, zstd
+func CreateTarArchive(src, dest, compression string, verbose, debug bool) error {
+	tarFile, err := os.Create(dest)
+	if err != nil {
+		return fmt.Errorf("Error creating tar archive \"%s\": \"%s\"", dest, err.Error())
+	}
+	var fileWriter io.WriteCloser = tarFile
+	tarWriter := tar.NewWriter(fileWriter)
+	defer tarWriter.Close()
+	return filepath.Walk(src, func(filePath string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// skip non-regular files that aren't symbolic links or directories
+		if !fileInfo.Mode().IsRegular() &&
+			!(fileInfo.Mode()&os.ModeSymlink == os.ModeSymlink) &&
+			!fileInfo.IsDir() {
```

Comment:
hm, wait. I was actually curious about what `IsRegular()` means in golang context, and I found this: "Please note that Executable are not regular even though in the UNIX tradition, they usually are". This seems to mean we will be skipping all executable files from the tarball?! I don't think that's intentional? Or is it not actually what's happening?

This also makes me think: what are the specific files we want to avoid fetching here? Is it the 'git related' ones?

Reply:
@jawn-smith What about my comment re: executables here? Did you have a chance to take a look at that?

Reply:
@sil2100 Yes, this should handle executables just fine. Not sure where you found that "Please note" section. However, per mwhudson's suggestion I'm rewriting the create/extract functions to call `usr/bin/tar`